exercises/scale-generator/scale_generator_test.exs

```diff
@@ -0,0 +1,276 @@
+if !System.get_env("EXERCISM_TEST_EXAMPLES") do
+  Code.load_file("scale_generator.exs", __DIR__)
+end
+
+ExUnit.start
+ExUnit.configure exclude: :pending, trace: true
+
+defmodule ScaleGeneratorTest do
+  use ExUnit.Case
+
+  @major_scale_pattern           "MMmMMMm"
+  @minor_scale_pattern           "MmMMmMM"
+  @dorian_scale_pattern          "MmMMMmM"
+  @mixolydian_scale_pattern      "MMmMMmM"
+  @lydian_scale_pattern          "MMMmMMm"
+  @phrygian_scale_pattern        "mMMMmMM"
+  @locrian_scale_pattern         "mMMmMMM"
+  @harmonic_minor_scale_pattern  "MmMMmAm"
```

Comment:
Part of me thinks that if we're going to include the harmonic minor scale, we should also include the melodic minor scale. They're frequently taught together, and both used frequently.

That pattern is "MmMMMMm", and an example based on C would be "C D Eb F G A B C".